packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -1273,4 +1280,57 @@ export class MirrorNodeClient {
     }
     return result;
   }
+
+  /**
+   * Retrieves and processes transaction record metrics from the mirror node based on the provided transaction ID.
+   *
+   * @param {string} transactionId - The ID of the transaction for which the record is being retrieved.
+   * @param {string} callerName - The name of the caller requesting the transaction record.
+   * @param {string} requestId - The unique identifier for the request, used for logging and tracking.
+   * @param {string} txConstructorName - The name of the transaction constructor associated with the transaction.
+   * @param {string} operatorAccountId - The account ID of the operator, used to calculate transaction fees.
+   * @returns {Promise<{transactionFee: number;} | undefined>} - An object containing the transaction fee if available, or `undefined` if the transaction record is not found.
+   */
+  public async getTransactionRecordMetrics(
+    transactionId: string,
+    callerName: string,
+    requestId: string,
+    txConstructorName: string,
+    operatorAccountId: string,
+  ): Promise<{ transactionFee: number } | undefined> {
+    const formattedRequestId = formatRequestIdMessage(requestId);
+
+    this.logger.trace(
+      `${formattedRequestId} Get transaction record via mirror node: transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}`,
+    );
+
+    const mirrorNodeRetries = parseNumericEnvVar(
+      'MIRROR_NODE_GET_CONTRACT_RESULTS_RETRIES',
+      'MIRROR_NODE_GET_CONTRACT_RESULTS_DEFAULT_RETRIES',
+    );
+
+    // poll mirror node to get transaction record
+    const transactionRecord = await this.repeatedRequest(
+      this.getTransactionById.name,
+      [transactionId, 0],
+      mirrorNodeRetries,
+      formattedRequestId,
+    );
+
+    if (!transactionRecord) {
+      this.logger.warn(
+        `${formattedRequestId} No transaction record retrieved: transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}`,
+      );
+    } else {
+      const transactionReceipt: IMirrorNodeTransactionRecord = transactionRecord.transactions.find(
```

Comment:
Ah yeah no worries! So the `transactionRecord` is actually the response from the `this.repeatedRequest()` method, which is located above at line 1314. After executing that line, `transactionRecord` will hold an array of transactions. From there, we traverse the array and use `.find()` to locate the specific transaction we want to work with. Let me know if it makes sense!

Reply:
Ah yes, for some reason i thought this else is when we DO NOT have a transaction record :d

Reply:
> Okay, then let's rename this interface because it's very misleading:
>
> IMirrorNodeTransactionRecord -> ITransaction
>
> and change the variable to:
>
> transactionReceipt -> transaction

@victor-yanev Thanks for the suggestion! However, I don’t see it as misleading. I specifically named it `IMirrorNodeTransactionRecord` to clearly indicate that it’s assigned to the transaction record response retrieved from querying the mirror node. This helps distinguish it from the transaction record response from the consensus node. Additionally, `ITransaction` feels a bit too general to me. That said, I do agree that `transactionReceipt` should be renamed, but I would rename it to `transactionRecord` instead of `transaction`.